### PR TITLE
Added cl_showping

### PR DESCRIPTION
**Pull request by Ryozuki**
It's same as fps but shows the ping (i get the ping from scoreboard ping)
I'm learning how to code in "teeworlds" xD, timakro is helping me :)

### CODE REVIEW

**Comment by def-** [comment]
Any other games do this or what's the idea behind this?

**Comment by Ryozuki** [comment]
As i have not so good internet, i like to know when i'm getting lag spykes without looking at scoreboard or control shit d. It's more clean.

File: src/game/server/player.cpp [hunk 1]
```diff
@@ -255,14 +255,16 @@ void CPlayer::Tick()
...
-		}
-	}
+if(m_PlayerFlags&PLAYERFLAG_SCOREBOARD)
+{
+ for(int i = 0; i < MAX_CLIENTS; ++i)
+ {
+	if(GameServer()->m_apPlayers[i] && GameServer()->m_apPlayers[i]->GetTeam() != TEAM_SPECTATORS)
+	 	m_aActLatency[i] = GameServer()->m_apPlayers[i]->m_Latency.m_Min;
+ }
+}
```
Comment (def-) [inline src/game/server/player.cpp]: You removed indentation here and in previous lines, please fix.

File: src/game/server/player.cpp [hunk 2]
```diff
@@ -315,6 +317,7 @@ void CPlayer::Snap(int SnappingClient)
 	}
 
 	CNetObj_PlayerInfo *pPlayerInfo = static_cast<CNetObj_PlayerInfo *>(Server()->SnapNewItem(NETOBJTYPE_PLAYERINFO, id, sizeof(CNetObj_PlayerInfo)));
+
```
Comment (def-) [inline src/game/server/player.cpp]: Don't insert empty lines in random locations.

**Comment by Ryozuki** [comment]
Ok i think now has good indentation.

**Comment by def-** [comment]
Nope, look at the diff here: https://github.com/ddnet/ddnet/pull/458/files

Also, you're mixing tabs and spaces there. And ideally choose more creative commit names than "Update $FILE.cpp"

**Comment by Henningstone** [comment]
Furthermore you may want to rebase it before it gets merged, so that there are not 5 commits which are basically trash...

File: src/game/client/components/hud.cpp [hunk 3]
```diff
@@ -319,6 +319,12 @@ void CHud::RenderFps()
 		str_format(Buf, sizeof(Buf), "%d", (int)m_AverageFPS);
 		TextRender()->Text(0, m_Width-10-TextRender()->TextWidth(0,12,Buf,-1), 5, 12, Buf, -1);
 	}
+
+	if(g_Config.m_ClShowping){
```
Comment (Henningstone) [inline src/game/client/components/hud.cpp]: Teeworlds code convention sais that the opening braces `{` of statements of all types goes to the next line

File: src/game/server/player.cpp [hunk 4]
```diff
@@ -257,12 +257,14 @@ void CPlayer::PostTick()
 	// update latency value
 	if(m_PlayerFlags&PLAYERFLAG_SCOREBOARD)
 	{
-		for(int i = 0; i < MAX_CLIENTS; ++i)
-		{
+ 		for(int i = 0; i < MAX_CLIENTS; ++i)
+ 		{
```
Comment (Henningstone) [inline src/game/server/player.cpp]: Spaces :\

**Comment by Ryozuki** [comment]
Fixed and rebased (i hope)

File: src/game/server/player.cpp [hunk 5]
```diff
@@ -260,9 +260,11 @@ void CPlayer::PostTick()
 		for(int i = 0; i < MAX_CLIENTS; ++i)
 		{
 			if(GameServer()->m_apPlayers[i] && GameServer()->m_apPlayers[i]->GetTeam() != TEAM_SPECTATORS)
-				m_aActLatency[i] = GameServer()->m_apPlayers[i]->m_Latency.m_Min;
+			m_aActLatency[i] = GameServer()->m_apPlayers[i]->m_Latency.m_Min;
```
Comment (def-) [inline src/game/server/player.cpp]: One tab missing here.

**Comment by def-** [comment]
Not sure how you got one of my commit s in there.

**Comment by Ryozuki** [comment]
I don't know how too (dafuq?).

I think now it's all fixed.

**Comment by def-** [comment]
Thanks.

**Comment by timakro** [comment]
The server did change too.

**Comment by Ryozuki** [comment]
You have to compile server

**Comment by def-** [comment]
So it only works on DDNet server, hm.

**Comment by Ryozuki** [comment]
searching for a solution.

**Comment by def-** [comment]
The only solution would be to set the SCOREBOARD flag in the client instead when cl_showping is activated, but that would increase the network bandwidth.

Instead you could just show the pred value, it's more important than the (made up) ping value in scoreboard anyway. pred value tells you how much delayed you see everything on server.

**Pull request merged**